### PR TITLE
Create provider for date-fns locale

### PR DESCRIPTION
**Pull request by jamesricky**
Currently only for use in DatePicker & DateRangePicker.

### CODE REVIEW

**Review by kaufmo — APPROVED** [review]

File: packages/admin/admin/src/contextProviders/DateFnsLocale.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { Locale } from "date-fns";
+import * as React from "react";
+
+export const DateFnsLocaleContext = React.createContext<Locale | undefined>(undefined);
```
Comment (johnnyomair) [inline packages/admin/admin/src/contextProviders/DateFnsLocale.tsx:4]: Shouldn't the context be in the date time package?

**Review by johnnyomair — APPROVED** [review]

**Comment by nsams** [comment]
I would have tried everything to avoid yet another provider, but I'll accept the decision you made

File: packages/admin/admin-date-time/src/DatePicker.tsx [hunk 2]
```diff
@@ -16,22 +22,38 @@ export interface DatePickerProps extends Omit<InputWithPopperProps, "children" |
...
+    maxDate?: Date;
+    minDate?: Date;
 }
 
-export function DatePicker({
+const defaultMinDate = new Date();
+defaultMinDate.setFullYear(defaultMinDate.getFullYear() - 15);
+
+const defaultMaxDate = new Date();
+defaultMaxDate.setFullYear(defaultMaxDate.getFullYear() + 15);
```
[the remark this answers is not in the document]
Comment (jamesricky) [reply packages/admin/admin-date-time/src/DatePicker.tsx:34]: No good reason, just visual reasons. 
I fixed the styling & changed the range (in #635) so now the currently oldest person alive can select their birthday 😄

**Comment by jamesricky** [comment]
> I would have tried everything to avoid yet another provider, but I'll accept the decision you made

@nsams I don't know of another way unfortunately... except the [previously mentioned](https://github.com/vivid-planet/comet-admin/pull/636#issuecomment-1142253984) hacky solution.

**Comment by johnnyomair** [comment]
Two options we have not considered (up to my knowledge)

1. Importing the locale via dynamic import based on react-intl's locale
2. An object with "common" locales (as shown [here](https://date-fns.org/v2.28.0/docs/I18n).

Both these solutions will only working if we restrict the date-fns locale imports to a certain amount of known locales. Presumably we only need to support German and English in most cases. However, we could support other locales on demand by adding a global `window.__LOCALES = { en: ...., de: ..., <language-in-app> }` which the picker components use.

**Review by nsams — APPROVED** [review]

**Comment by johnnyomair** [comment]
@jamesricky please resolve merge conflicts and add a changelog entry

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**